src/CoherenceBuild/PackagePublisher.cs

```diff
@@ -46,54 +45,65 @@ public static void ExpandPackageFiles(IEnumerable<PackageInfo> processedPackages
 
         private static async Task PublishToFeedAsync(IEnumerable<PackageInfo> processedPackages, string feed, string apiKey)
         {
-            var packagesToPush = processedPackages.OrderBy(a => a.Degree);
             using (var semaphore = new SemaphoreSlim(4))
             {
                 var sourceRepository = CreateSourceRepository(feed);
 
                 var metadataResource = await sourceRepository.GetResourceAsync<MetadataResource>();
                 var packageUpdateResource = await sourceRepository.GetResourceAsync<PackageUpdateResource>();
-                var tasks = processedPackages.Select(async package =>
+
+                // Group packages to push by degree.
+                var packageGroups = processedPackages.GroupBy(p => p.Degree).OrderBy(g => g.Key);
```

Comment:
This gets us the same sort of push order as our build order in Coherence. The hope is that this pushes packages so we don't have transient restore issues.